test/functional/test_framework/key.py

```diff
@@ -384,3 +453,33 @@ def sign_ecdsa(self, msg, low_s=True):
         rb = r.to_bytes((r.bit_length() + 8) // 8, 'big')
         sb = s.to_bytes((s.bit_length() + 8) // 8, 'big')
         return b'\x30' + bytes([4 + len(rb) + len(sb), 2, len(rb)]) + rb + bytes([2, len(sb)]) + sb
+
+    def sign_schnorr(self, msg):
+        """Construct a bip-schnorr compatible signature with this key."""
+        assert(self.valid)
+        assert(self.compressed)
+        assert(len(msg) == 32)
+        x = self.secret
+        pk = self.get_pubkey()
+        if not pk.is_positive:
+            x = SECP256K1_ORDER - x
+        kp = int.from_bytes(TaggedHash("BIPSchnorrDerive", x.to_bytes(32, 'big') + msg), 'big') % SECP256K1_ORDER
```

Comment:
```suggestion
        kp = int.from_bytes(TaggedHash("BIP340Derive", x.to_bytes(32, 'big') + msg), 'big') % SECP256K1_ORDER
```

Not sure if this breaks anything... :confused: